File: app/src/main/java/com/ericho/coupleshare/mvp/presenter/PhotoPresenter.java
package com.ericho.coupleshare.mvp.presenter;

import com.ericho.coupleshare.R;
import com.ericho.coupleshare.mvp.Photo;
import com.ericho.coupleshare.mvp.PhotosContract;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by steve_000 on 17年6月25日.
 * for project CoupleShare
 * package name com.ericho.coupleshare.mvp.presenter
 */

public class PhotoPresenter implements PhotosContract.Presenter {

    PhotosContract.View mView;


    public  PhotoPresenter(PhotosContract.View view){
        this.mView = view;
    }


    @Override
    public void start() {
        loadPhotos();
    }

    @Override
    public void loadPhotos() {
        mView.showPhotoList(dummyPhotoList());
    }

    @Override
    public void addNewPhotos() {

    }


    private List<Photo> dummyPhotoList(){
        List<Photo> res = new ArrayList<>();
        Photo p =new Photo();
        p.setImageInt(R.drawable.ic_menu_manage);
        res.add(p);
        Photo p2 =new Photo();
        p.setImageInt(R.drawable.ic_menu_camera);
        res.add(p2);
        return res;
    }
}

File: app/src/main/java/com/ericho/coupleshare/mvp/presenter/PhotoPresenter.kt
package com.ericho.coupleshare.mvp.presenter

import com.ericho.coupleshare.mvp.PhotoBo
import com.ericho.coupleshare.mvp.PhotosContract
import java.util.ArrayList

/**
 * Created by steve_000 on 4/7/2017.
 * for project CoupleShare
 * package name com.ericho.coupleshare.mvp.presenter
 */
class PhotoPresenter( view:PhotosContract.View) :PhotosContract.Presenter{


    private val mView:PhotosContract.View

    init{
        mView = requireNotNull(view)
    }

    override fun start() {

    }

    override fun loadPhotos() {
//        mView.showPhotoList(dummyPhotoList())
    }

    override fun addNewPhotos() {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    private fun dummyPhotoList(): List<PhotoBo> {
        val res = ArrayList<PhotoBo>()
        val p = PhotoBo()
        p.suffixUrl = ""
        res.add(p)
        val p2 = PhotoBo()
        res.add(p2)
        return res
    }
}